src/librustc_errors/emitter.rs

```diff
@@ -703,6 +703,21 @@ impl EmitterWriter {
         }
     }
 
+    fn msg_with_padding(&self, msg: &str, padding: usize) -> String {
+        let padding = (0..padding)
+            .map(|_| " ")
+            .collect::<String>();
+        msg.split('\n').enumerate().fold("".to_owned(), |mut acc, x| {
+            if x.0 != 0 {
+                // Align every line with first one.
+                acc.push_str(&padding);
+            }
+            acc.push_str(&x.1);
+            acc.push_str("\n");
+            acc
```

Comment:
I need other opinions, but I think that before 711 lines you call measure a new string size and reserve it to prevent allocations in 713, 715, 716 lines.